### DISCUSSION TITLE
get container logs: "tail" parameter not valid

### DISCUSSION THREAD

**Discussion by sannaroby** [testcontainers/testcontainers-node#722] [Q&A] [answered]
We're using testcontainers-node library (v10.7.2) to launch [localstack](https://github.com/localstack/localstack/releases/tag/v3.2.0) (v3.2.0-pro) container in our integration tests.
We launch the integration tests in our ci environment and we use [kubedock](https://github.com/joyrex2001/kubedock) to orchestrate the containers.

The localstack container starts, but it is stopped immediately afterwards. It seems the problem depends on the testcontainers-node library that is sending a not valid "tail" parameter for the GET container logs requests.

These are the log retrieved from the kubedock container:

```
2024-03-04T21:21:40.114733812Z I0304 21:21:40.114582      1 logs.go:72] error retrieving logs: PodLogOptions "kubedock-e19768203586" is invalid: tailLines: Invalid value: -1: must be greater than or equal to 0

2024-03-04T21:21:40.114733812Z I0304 21:21:40.114582       1 util.go:100] Request: 
[GIN] 2024/03/04 - 21:21:40 | 200 |    6.837861ms |       127.0.0.1 | GET      "/containers/e197682035869ca70a0b5b2c839f3d80727a557fb8edf5dddf6c4fa27c86aa08/logs?follow=true&stdout=true&stderr=true&tail=-1&since=0"

```

After this error, the localstack container is stopped.

Is it possibile for you to avoid to include the tail parameter if it is not defined ? Or there is a way for me to avoid the problem ?

Thanks for your support.

**Comment by cristianrgreco** [comment, marked as answer]
Hi @sannaroby, the [Docker log docs](https://docs.docker.com/reference/cli/docker/container/logs/#description) say:
> Passing a negative number or a non-integer to --tail is invalid and the value is set to all in that case.

I'm not familiar with kubedock but it seems it doesn't behave the same way. Perhaps a PR to change the default of `-1` to `all` (https://github.com/NuShoSinkuPomogliTebeTvoiSankcii/testcontainers-node/blob/cb56edabcc8e686e72da4a02f05120b1f021a398/packages/testcontainers/src/container-runtime/clients/container/docker-container-client.ts#L165) would do it. 

This was added recently, so as a workaround version 10.6.0 should work for you.

**Reply by sannaroby** [reply]
Thanks @cristianrgreco for your help. I solved using the 10.6.0 version in my tests. Furthermore, the authors of "kubedock" also fixed the component in order to avoid a crash with a not valid "tail" parameter
